### ISSUE TITLE
ITK-snap not launching in 1.7.3

### ISSUE THREAD

**Issue by saimarathore** [CBICA/CaPTk#743]
ITK-snap not launching on my machine in 1.7.3, whereas it works fine in previous versions (like 1.6.2).
Windows 7
Desktop
CaPTk version: 1.7.3

**Comment by ashishsingh18** [comment]
@saimarathore  I am not able to replicate this issue. ITK-Snap launches for me on version 1.7.3

**Comment by ashishsingh18** [comment]
@dboun @saimarathore @sarthakpati I am not able to replicate this issue. Can you all please verify if this is replicable? If not we close it as 'not reproducible'.

**Comment by sarthakpati** [comment]
Cannot replicate on 1.7.3 on multiple Windows (7 & 10) machines.

**Comment by dboun** [comment]
Linux Mint 19.2:

- Pressing itksnap without images -> it says please load some images
- Pressing itksnap with images -> it says mask file has no pixels greater than 0
- Pressing itksnap with some drawing on the ROI image -> it says 'Mask saved from ITK-SNAP loaded' (itk-snap never opened)

**Comment by dboun** [comment]
ITK-SNAP doesn't work on Ubuntu 18.04 and derivatives because it requires a system library called ```libpng12.so``` that has since been deprecated (it was part of 16.04 or something) [[source]](https://askubuntu.com/a/895903)

```bash
$ ./itksnap
/home/dimitris/Work/CaPTk/CaPTk-build-ext-qt/Debug/externalApps/itksnap/lib/snap-3.8.0/ITK-SNAP: error while loading shared libraries: libpng12.so.0: cannot open shared object file: No such file or directory

```

**Comment by MarkBergman-cbica** [comment]
itk-snap does not launch due to incorrect permissions:


     1  (base) [user@login4 ~]$ which captk
     2  /software/lab/captk/centos7/1.7.2/bin/captk
     3  (base) [user@login4 ~]$ module unload captk/1.7.2 
     4  (base) [user@login4 ~]$ module load captk/1.7.6
     5  (base) [user@login4 ~]$ which captk
     6  /software/lab/captk/centos7/1.7.6/bin/captk
     7  (base) [user@login4 ~]$ which itksnap 
     8  /software/lab/captk/centos7/1.7.6/bin/itksnap
                Good

     9  (base) [user@login4 ~]$ itksnap
    10  Error running "/software/lab/captk/centos7/1.7.6/lib/snap-3.8.0/ITK-SNAP": Permission denied
                Bad.

                Under Linux it is unusual to have a package installed into a sub-directory called lib.

    11  (base) [user@login4 ~]$ ls -l /software/lab/captk/centos7/1.7.6/lib/snap-3.8.0/ITK-SNAP
    12  -rw-r--r--. 1 adminuser admingrp 78066091 Nov 27 22:31 /software/lab/captk/centos7/1.7.6/lib/snap-3.8.0/ITK-SNAP
                The file does not have executable permission -- this may be corrected in packaging or in our installer, after expanding itk-snap.

    13  (base) [user@login4 ~]$ ls -l /software/lab/captk/centos7/1.7.6/bin/itksnap
    14  -rwxr-xr-x. 1 adminuser admingrp 78079 Nov 27 22:31 /software/lab/captk/centos7/1.7.6/bin/itksnap*
                OK, so the end-user executable has the correct permissions, but the target that it exec()s does not.

**Comment by dboun** [comment]
@MarkBergman-cbica @sarthakpati 

I downloaded itk-snap from its website and the files are executable from scratch. So we just need to add their latest code in our binaries zip (**but not do it from windows**, zip preserves permissions, but when you unzip it on windows they get removed). I don't think we need any additional complex fix like the above commits. 

About the libpng12.so issue that I faced, there is an issue on itk-snap about it since Oct 2019 [[LINK]](https://github.com/pyushkevich/itksnap/issues/5). In [that comment specifically](https://github.com/pyushkevich/itksnap/issues/5#issuecomment-545076690) paul provides a modern build on debian 10. If you are interested we can switch to that.
